### DISCUSSION TITLE
JIRA integration

### DISCUSSION THREAD

**Discussion by dabeeeenster** [Flagsmith/flagsmith#2133] [Ideas]
Several customers have discussed/requested a JIRA integration. JIRA is a large product! This discussion is to go over what potential functionality an Integration could entail. 

## View/Modify flags within JIRA

JIRA allows you to surface third party UI similar to our [Datadog dashboard widget](https://docs.flagsmith.com/integrations/apm/datadog) integration. We can allow for the following actions:

### Create a flag and associate it with a JIRA issue

Within the JIRA UI, navigate to an issue. Then create a new flag. Provide the default state. This would associate the flag with the currently selected JIRA issue. 

### Associate an existing flag to an existing JIRA issue

Within the JIRA UI, create a new flag. Provide the default state. Associate it with the currently selected JIRA issue. 

### Modify the state of the flag in Jira

Given the appropriate permissions, allow the user to modify the flag state within Jira for a specific environment.

## Further Info

Atlassian have a [Marketplace](https://marketplace.atlassian.com/) that could house the integration.

**Comment by dabeeeenster** [comment]
Implemented! https://docs.flagsmith.com/integrations/project-management/jira

**Comment by henninglube** [comment]
Awesome thanks a lot.
